src/prop/prop_proof_manager.cpp

```diff
@@ -338,9 +340,70 @@ Node PropPfManager::normalizeAndRegister(TNode clauseNode,
   {
     d_satPm->registerSatAssumptions({normClauseNode});
   }
+  // if proof logging, make the call now
+  if (d_plog != nullptr)
+  {
+    if (!input)
+    {
+      if (d_env.isTheoryProofProducing())
+      {
+        // if theory proof producing, we get the proof to log
+        std::shared_ptr<ProofNode> pfn = d_proof.getProofFor(normClauseNode);
+        d_plog->logTheoryLemmaProof(pfn);
+      }
+      else
+      {
+        // otherwise we just notify the clause
+        d_plog->logTheoryLemma(normClauseNode);
+      }
+    }
+  }
   return normClauseNode;
 }
 
+void PropPfManager::presolve()
+{
+  // get the proof logger now
+  d_plog = d_env.getProofLogger();
+  Trace("pf-log-debug") << "PropPfManager::presolve, plog="
+                        << (d_plog != nullptr) << std::endl;
+  if (d_plog != nullptr)
+  {
+    // TODO (wishues #157): in incremental mode, only get the new assertions
+    std::vector<std::shared_ptr<ProofNode>> icp = getInputClausesProofs();
+    for (const Node& a : d_assumptions)
+    {
+      icp.emplace_back(d_proof.getProofFor(a));
+    }
+    Trace("pf-log-debug") << "PropPfManager::presolve, we have "
+                          << d_inputClauses.size() << " inputs and "
+                          << d_assumptions.size() << " assumptions"
+                          << std::endl;
+    d_plog->logCnfPreprocessInputProofs(icp);
+  }
+}
+
+void PropPfManager::postsolve(SatValue result)
+{
+  if (d_plog != nullptr)
+  {
+    if (result == SAT_VALUE_FALSE)
```

Comment:
A high-level comment: what happens here is that we log whatever SAT proof was generated, if any. In principle, and what would be more in the spirit of proof logging in SAT solvers, we'd be logging the learned SAT clauses as they are generated. Do I understand correctly that you did not go this route because that'd require instrumenting the SAT solvers to do this and you want to have the infrastructure independent of the SAT solver?

Reply:
We could in theory interleave the SAT proof logging with the theory lemma logging.  This call would then just serve as a "final notification" to print whatever is needed to prove false, or maybe do nothing at all if it was already logged.

Reply:
Thinking more about this, I think for the intended usage of the proof logger this interleaving is what makes more sense. But I think it's fine to have this in with the current behavior and later refined.